### PR TITLE
Correct htlc witness size calculations

### PR DESCRIPTION
**Pull request by halseth**
I noticed a few inconsistencies in the weight constants. 

I guess implementations just use the hardcoded constants for `HTLC-timeout` and `HTLC-success` when doing tx constructions (this is what lnd does), so I added a note here that we should continue using this value even though it is off by one.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Is this an artifact of signature size variation? I mean there is no "right" answer after you sign things vary a byte or two, but is the limit wrong because we didn't account for low-S implying signatures are never max size?

**Comment by halseth** [comment]
We always assume they are the worst case size (73). Are you saying signatures never actually get that large? Interesting.

**Comment by t-bast** [comment]
Wow good catch, I don't understand why we considered this `cltv_expiry` to be encoded with 3 bytes (and how the `success_witness` turned out to be incorrect...). Time for some git speleology

File: 03-transactions.md [hunk 1]
```diff
@@ -954,11 +954,12 @@ The *expected weight* of an HTLC transaction is calculated as follows:
...
 
 Multiplying non-witness data by 4 results in a weight of 376. Adding
-the witness data for each case (285 or 288 + 2 for HTLC-timeout, 325 or 328 + 2 for
+the witness data for each case (285 or 288 + 2 for HTLC-timeout, 324 or 327 + 2 for
 HTLC-success) results in weights of:
 
 	663 (HTLC-timeout) (666 with with option_anchor_outputs))
-	703 (HTLC-success) (706 with with option_anchor_outputs))
+	703 (HTLC-success) (706 with with option_anchor_outputs)) 
+                - (reeeally 702 and 705, but we use these numbers for historical reasons)
```
Comment (cdecker) [inline 03-transactions.md:962]: 😂

**Review by t-bast — APPROVED** [review]

**Comment by t-bast** [comment]
I re-did the calculations and I agree with what you found.
Can someone else check again to keep me honest?

File: 03-transactions.md [hunk 2]
```diff
@@ -954,11 +954,12 @@ The *expected weight* of an HTLC transaction is calculated as follows:
 		- lock_time: 4 bytes
 
 Multiplying non-witness data by 4 results in a weight of 376. Adding
-the witness data for each case (285 or 288 + 2 for HTLC-timeout, 325 or 328 + 2 for
+the witness data for each case (285 or 288 + 2 for HTLC-timeout, 324 or 327 + 2 for
 HTLC-success) results in weights of:
 
 	663 (HTLC-timeout) (666 with with option_anchor_outputs))
-	703 (HTLC-success) (706 with with option_anchor_outputs))
+	703 (HTLC-success) (706 with with option_anchor_outputs))
```
Comment (flack) [inline 03-transactions.md:961]: `with with` is probably a typo, right?

Comment (halseth) [reply 03-transactions.md]: most certainly

**Review by t-bast — APPROVED** [review]
ACK https://github.com/lightningnetwork/lightning-rfc/pull/815/commits/2a8da1a3ef06f224a296c98e22c94db25ea41506

**Pull request merged**